### DISCUSSION TITLE
YAML metadata to trigger custom functions that populate missing values

### DISCUSSION THREAD

**Discussion by cueckoo** [cue-lang/cue#1069] [Q&A]
_Originally opened by @benbooth493 in https://github.com/cuelang/cue/discussions/1069_

Let’s say I have a yaml file with some values that can be empty.
Now say I have a program that reads the yaml and populates some of the values.
The values to populate is determined by a json schema document with `additionalProperties: true` and with some of the values having a custom property `mustPopulate: true`.

I am not using the json schema for validation (although I could of course), it’s really just to carry some additional metadata for the yaml file.

Is this pattern possible with Cue? If so, how would I do such a thing?

**Comment by myitcv** [comment]
@benbooth493 apologies for the delay in replying to this. We were busy completing our migration to a new home (#1078).

Is this still a requirement for you? If so, can you provide some more details with sample data perhaps?
